src/components/Layout/NavigationLayout/GenreNavigation/GenreNavigation.tsx

```diff
@@ -22,16 +39,33 @@ const GenreNavigation = (): JSX.Element => {
         {option.id}
       </button>
     ));
+
+  function updateFilters(newFilters): void {
+    queryParams.delete(FILTER_BY);
+    queryParams.append(FILTER_BY, newFilters);
+    history.replace({
+      search: queryParams.toString(),
+    });
+    setFilters(newFilters);
+  }
+
   const handleMenuItemClick = (event): void => {
+    let newFilters;
     const elementIndex = filters.indexOf(event.target.value);
     if (event.target.value === ALL_FILTERS) {
```

Comment:
```suggestion
    if (event.target.value === ALL_FILTERS) {
        newFilters = [];
    } else if (elementIndex === -1) {
        newFilters = filters.concat([event.target.value]);
    } else {
        newFilters = filters.filter((filter) => filter !== event.target.value);
    }
    dispatch(moviesActions.setFilter(newFilters));
    updateFilters(newFilters);
```
Just to remove repetition of the last 2 lines

Reply:
yes, thanks